modules/ip6/datapath/ip6_local.c

```diff
@@ -77,17 +77,25 @@ static uint16_t ip6_input_local_process(
 		if (edge == UNKNOWN_PROTO)
 			goto next;
 
-		// verify checksum if not already checked by hardware
-		switch (m->ol_flags & RTE_MBUF_F_RX_L4_CKSUM_MASK) {
-		case RTE_MBUF_F_RX_L4_CKSUM_NONE:
-		case RTE_MBUF_F_RX_L4_CKSUM_UNKNOWN:
-			if (rte_ipv6_udptcp_cksum_verify(
-				    ip, rte_pktmbuf_mtod_offset(m, void *, l3_hdr_size)
-			    ))
+		switch (d->proto) {
+		case IPPROTO_ICMPV6:
+		case IPPROTO_TCP:
+		case IPPROTO_UDP:
+			// verify checksum if not already checked by hardware
+			switch (m->ol_flags & RTE_MBUF_F_RX_L4_CKSUM_MASK) {
+			case RTE_MBUF_F_RX_L4_CKSUM_NONE:
+			case RTE_MBUF_F_RX_L4_CKSUM_UNKNOWN:
+				if (rte_ipv6_udptcp_cksum_verify(
+					    ip, rte_pktmbuf_mtod_offset(m, void *, l3_hdr_size)
+				    )) {
+					edge = BAD_CHECKSUM;
+					goto next;
```

Comment:
Nit: those jumps to next seem unneeded.

Reply:
These were a remnant of poorly ordered patches. It should make more sense now.